smarc_simulated_comms/scripts/comms_master.py

```diff
@@ -0,0 +1,132 @@
+#! /usr/bin/env python3
+# -*- coding: utf-8 -*-
+# vim:fenc=utf-8
+#
+# Author: Ozer Ozkahraman (ozkahramanozer@gmail.com)
+# Date: 2018-04-11
+
+import time
+import rospy
+import random
+import math
+
+from gazebo_msgs.msg import ModelStates
+from smarc_msgs.msg import CommsMessage
+
+MAX_DIST = 20
+RESCAN_PERIOD = 10
+
+
+class CommsMaster:
+    def __init__(self):
+        # know all the positions of the models
+        rospy.Subscriber('/gazebo/model_states', ModelStates, self._CB_model_states)
+        self.model_states = None
+
+        # listen to incoming packets
+        rospy.Subscriber('/comms/inbound', CommsMessage, self._CB_packet_received)
+
+
+        # a list of known communicators. 
+        # there should be 'namespace:publisher' pairs in here
+        self.known_agents = {}
+        self._scan_topics()
+
+        global MAX_DIST
+        self.MAX_DIST = MAX_DIST
+        global RESCAN_PERIOD
+        self.RESCAN_PERIOD = RESCAN_PERIOD
+        self._last_update = 0
+
+
+    def update(self):
+        if rospy.Time.now() - self._last_update > self.RESCAN_PERIOD:
+            self._scan_topics()
+            self._last_update = rospy.Time.now()
+
+    def _scan_topics(self):
+        """
+        this looks at all topics and tries to find lolos and sams
+        """
+        target_names = ['lolo', 'sam']
```

Comment:
Please make this configurable with a ROS parameter

Reply:
I have found no way of reading in a list of parameters.

Reply:
See e.g. here : https://github.com/smarc-project/smarc_navigation/blob/master/auv_ekf_slam/src/ekf_slam.cpp#L18 and https://github.com/smarc-project/smarc_navigation/blob/master/auv_ekf_slam/launch/ekf_slam.launch#L27

Reply:
done